### DISCUSSION TITLE
Bug: Infinite loop calling requestAnimationFrame function

### DISCUSSION THREAD

**Discussion by askel4dd** [twilio-labs/paste#2281] [General] [answered]
Hi team! Is it me or Paste is doing something weird? I feel like Firefox cannot handle that many requestAnimationFrame per milliseconds, it’s not normal. Stack trace is pointing towards this [Paste package](https://github.com/twilio-labs/paste/blob/1bcb8b3093920fdd871f8a13b498a8eb99201200/packages/paste-libraries/animation/src/index.tsx) and [this library](https://github.com/pmndrs/react-spring). I found this [issue](https://github.com/pmndrs/react-spring/issues/1002), it might be related

<img width="1792" alt="Screenshot 2022-03-09 at 14 53 00" src="https://user-images.githubusercontent.com/7442658/157708710-32d1b80d-e365-49a5-b5cd-df2bfa63c339.png">

**Comment by zahnster** [comment, marked as answer]
Noticed this resolved issue with React Spring that looks to be the root cause of this issue: https://github.com/pmndrs/react-spring/pull/1786

When I updated our dependency locally, I'm no longer seeing the RAF overload. Will put up a PR shortly.

**Reply by nkrantz** [reply]
PR in case you want to keep track, @askel4dd: https://github.com/twilio-labs/paste/pull/2282

**Comment by zahnster** [comment]
Hey @askel4dd,

We were able to work through the blocking issues and the update is now available with Paste Core v12.1.0. Thanks for your patience while we worked on this, and for bringing this to our attention!
